services/src/main/java/com/linkedin/kmf/services/ProduceService.java

```diff
@@ -97,26 +97,16 @@ public ProduceService(Map<String, Object> props, String name) throws Exception {
       }
     }
 
-    double partitionsToBrokersRatio = config.getDouble(CommonServiceConfig.PARTITIONS_TO_BROKER_RATO_CONFIG);
-    int topicReplicationFactor = config.getInt(ProduceServiceConfig.TOPIC_REPLICATION_FACTOR_CONFIG);
     int existingPartitionCount = Utils.getPartitionNumForTopic(_zkConnect, _topic);
-
-    if (existingPartitionCount <= 0) {
-      if (config.getBoolean(ProduceServiceConfig.TOPIC_CREATION_ENABLED_CONFIG)) {
-        _partitionNum.set(
-            Utils.createMonitoringTopicIfNotExists(_zkConnect, _topic, topicReplicationFactor,
-                partitionsToBrokersRatio));
-      } else {
-        throw new RuntimeException("Can not find valid partition number for topic " + _topic +
-            ". Please verify that the topic \"" + _topic + "\" has been created. Ideally the partition number should be" +
-            " a multiple of number" +
-            " of brokers in the cluster.  Or else configure " + ProduceServiceConfig.TOPIC_CREATION_ENABLED_CONFIG +
-            " to be true.");
-      }
-    } else {
-      _partitionNum.set(existingPartitionCount);
+    int topicPollDelay = config.getInt(ProduceServiceConfig.TOPIC_POLL_DELAY_CONFIG);
+    while (existingPartitionCount <= 0) {
```

Comment:
Neither start() nor the constructor should block waiting for topic to be created.